pynav/lib/camera.py

```diff
@@ -0,0 +1,219 @@
+"""Module containing a basic pinhole camera model."""
+
+import numpy as np
+from pynav.lib import SE3State
+
+
+class CameraModel:
+    """Class for a pinhole camera model.
+
+    This class contains utilities for generating measurements
+    and predicting measurements, and converting pixel coordinates to
+    normalized image coordinates using the camera intrinsic parameters.
+    """
+
+    def __init__(
+        self,
+        fu: float,
+        fv: float,
+        cu: float,
+        cv: float,
+        image_width: int,
+        image_height: int,
+        sigma: float,
+        T_bc: SE3State = None,
```

Comment:
I agree that the way the SE3State here is being used doesn't really utilize any of the features of the State object and it might make more sense to just have this be a 4x4 Numpy array. That being said, I do like the convenient getters for attitude and position...Also, the extrinsics are often added as a state for estimation in a lot of SLAM algorithms, but you're right that on the modelling side of things there might be no reason to have this be a state

Reply:
at the very least, i vote we make the user provide the matrix. then convert to SE3State for those getters if you really want. but these getters really arnt saving that much code

Reply:
to me, from the calling side, it kinda supplies that something fancy/dynamic is going to happen when really it is just a static parameter, for now